R/init_app.R

```diff
@@ -0,0 +1,161 @@
+#' Initialize the DFE Shiny Template Setup
+#'
+#' This function sets up a complete DFE Shiny template by:
+#' 1. Cloning the Shiny template repository.
+#' 2. Installing a Git pre-commit hook.
+#' 3. Setting up a GitHub Actions workflow for deployment.
+#'
+#' @param dashboard_name A character string specifying the dashboard name for the deployment configuration.
+#' If NULL, defaults to `"dfe-shiny-template"`.
+#'
+#' @details This function runs `init_app()`, `init_commit_hooks()`, and `init_workflow()`
+#' in sequence to set up a working environment for a Shiny application.
+#'
+#' @return No return value. The function initializes the complete template.
+#' @examples
+#' \dontrun{
+#' init_template()
+#' init_template(dashboard_name = "my-dashboard")
+#' }
+#' @export
+init_template <- function(dashboard_name = "dfe-shiny-template") {
+  message("Initializing the DFE Shiny template...")
+
+  # Step 1: Clone the Shiny template repository
+  init_app()
+
+  # Step 2: Install Git pre-commit hooks
+  init_commit_hooks()
+
+  # Step 3: Setup GitHub Actions workflow
+  init_workflow(dashboard_name = dashboard_name)
```

Comment:
Either as extra functions or part of init_workflow, I'd like to see the other workflows created as well:
- automated tests (shinytest2)
- tidy code